src/CalcViewModel/Common/NavCategory.cpp

```diff
@@ -48,45 +47,57 @@ static constexpr int CURRENCY_ID = 16;
 static constexpr int GRAPHING_ID = 17;
 // ^^^ THESE CONSTANTS SHOULD NEVER CHANGE ^^^
 
-wchar_t* towchar_t(int number)
+namespace
 {
-    auto wstr = to_wstring(number);
-    return _wcsdup(wstr.c_str());
-}
-
-bool IsGraphingModeAvailable()
-{
-    static bool supportGraph = Windows::Foundation::Metadata::ApiInformation::IsMethodPresent("Windows.UI.Text.RichEditTextDocument", "GetMath");
-    return supportGraph;
-}
+    Platform::Agile<Windows::System::User^> CurrentUser;
+    std::mutex GraphingModeCheckMutex;
 
-Box<bool> ^ _isGraphingModeEnabledCached = nullptr;
-bool IsGraphingModeEnabled(User ^ currentUser = nullptr)
-{
-    if (!IsGraphingModeAvailable())
+    wchar_t* towchar_t(int number)
```

Comment:
We should remove this function, since it's allocating memory which is never cleaned up. One way to fix this would be to  call `std::to_wstring(number).c_str()` inline at this function's call sites.

Reply:
Good point. Yes, `towchar_t` can be remove.
I'd prefer changing the type of `NavCategoryInitializer::accessKey` from `const wchar_t*` into `std::optional<std::wstring>` so that we can manage the memory and keep its semantics unchanged as well.